app/controllers/mixins/compressed_ids.rb

```diff
@@ -1,6 +1,7 @@
 module CompressedIds
+  CID_REGEXP = '[0-9]r?[0-9]*'.freeze
```

Comment:
In some cases we cannot reuse it one to one. That is because we need one Regexp to parse numeric items *from* cid, then we need another regexp to parse cid from url.

What I can do is to have both reg-exps in ArRegion side by side.